### ISSUE TITLE
Check gradient is not correct with CrossEntropy as loss measure.

### ISSUE THREAD

**Issue by EmanueleLM** [EmanueleLM/DeepNet#3]
No matter if you use a fully connected topology or a non-fully connected, the check gradient is incorrect, still the net works properly and performs better on problems when the cross entropy is better than L1,L2 (e.g. the 8*8 handwritten digits recognition)

**Comment by EmanueleLM** [comment]
CrossEntropy is not differentiable in all of its domain, as some activations such as relu and leakyrelu are.
Issue closed and modified the values the CrossEntropy assumes in the critical regions from magnitude 1e-4 to 1e-10 (check derivatives.py and loss.py)
